ui/shared/ad/CustomAdBanner.tsx

```diff
@@ -0,0 +1,73 @@
+import { Flex, chakra, Tooltip, Image } from '@chakra-ui/react';
+import { useQuery } from '@tanstack/react-query';
+import React, { useState, useEffect } from 'react';
+
+import type { AdCustomConfig } from 'types/client/ad';
+
+import config from 'configs/app';
+import type { ResourceError } from 'lib/api/resources';
+import { MINUTE } from 'lib/consts';
+import useFetch from 'lib/hooks/useFetch';
+import useIsMobile from 'lib/hooks/useIsMobile';
+
+const CustomAdBanner = ({ className }: { className?: string }) => {
+  const isMobile = useIsMobile();
+
+  const feature = config.features.adsBanner;
+  const configUrl = (feature.isEnabled && feature.provider === 'custom') ? feature.configUrl : '';
+
+  const apiFetch = useFetch();
+  const { data: adConfig } = useQuery<unknown, ResourceError<unknown>, AdCustomConfig>(
```

Comment:
I think we need to manage the following states of the component as well:
- loading - while the config is loading it is good to show a skeleton
- error - if the request fails then render nothing (`null`)